spec/controllers/v2/admin_controller_spec.rb

```diff
@@ -7,6 +7,12 @@
 
       subject { get :users }
 
+        context 'when data is invalid' do
+          subject { get :users, params: { id: (-1) } }
```

Comment:
You need to rescue from error and manually respond with 404

Reply:
We cannot of course anticipate all the errors. But we need to anticipate the users or hackers to try to play with us. 

So in this particular case we need to rescue from ActiveRecord::RecordNotFound.

It is needed for the backend developers to support communication with frontend using http statuses

Reply:
And 400 is not 404 :) 

Oh and now I saw why this is not needed. It is cause you are passing the params, but not using them :D

Reply:
Okej, musze to napisac na chwilkę po polsku bo sie pogubie :D. Rozumiem, ze przekazuje parametry pod zmienną subject  (ktora jest glownym obiektem testu). Więc skoro tylko ją przekazuję, to jak ją potem wywołać? Wiem, glupie pytanie, ale  trudno mi to znaleźć w internecie. Aktualnie po pokombinowaniu z let to test mi pokazuje ze nie może create_list z identycznymi id=-1 bo to pole jest unique - dalej źle, ale w inny sposób XD

Reply:
Aby wywołać subject wystarczy tylko go użyć w bloku 'it'

A tutaj chodzi o to by przekazać id od obiektu który nie istnieje. Więc potrzeba tylko dodać obsługę RecordNotFound w kontrolerze.

Reply:
OKEY, so: I added new action for V2::AdminController, it is 'show_user'. And there it works - if we type an 'id' that does not exisst (for ex, 2) it show 404 User not found. The test are okey. I am wondering if we have to test 'invalid params id=-1' also for method 'users' which is an index-like method. (because in url the is no ':id') :)) waiting for feedback <3